inst/include/dust/interface.hpp

```diff
@@ -371,6 +371,53 @@ void dust_reorder(SEXP ptr, cpp11::sexp r_index) {
   obj->reorder(index);
 }
 
+template <typename T>
+SEXP dust_resample(SEXP ptr, cpp11::doubles r_weights) {
+  Dust<T> *obj = cpp11::as_cpp<cpp11::external_pointer<Dust<T>>>(ptr).get();
+  size_t n_particles = obj->n_particles();
+  size_t n_pars = obj->n_pars();
+  size_t n_particles_each = n_particles;
+  if (n_pars == 0) {
+    if (static_cast<size_t>(r_weights.size()) != n_particles) {
+      cpp11::stop("Expected a vector with %d elements for 'weights'",
+                  n_particles);
+    }
```

Comment:
Any other checks on weights needed?:

- Must they sum to 1?
- At least one weight > 0? 
- No negative weights?

Reply:
certainly no negative. I will check that we do something sensible with no positive weights though. when we hand over to the C++ for this there will be no checking or error handling....